android/src/main/java/com/globo/reactnativeua/ReactNativeUA.java

```diff
@@ -156,4 +140,53 @@ private boolean shouldRequestPermissions(Activity activity) {
 
         return corseLocation == PackageManager.PERMISSION_DENIED && fineLocation == PackageManager.PERMISSION_DENIED;
     }
+
+    @ReactMethod
+    public void setAndroidSmallIcon(String iconName) {
+        int iconId = getImageResourceId(iconName);
+        if (iconId > 0) {
+            Preferences.getInstance().setAndroidSmallIconResourceId(iconId);
+            DefaultNotificationFactory defaultNotifFactory = getDefaultNotificationFactory();
+            defaultNotifFactory.setSmallIconId(iconId);
+            UAirship.shared().getPushManager().setNotificationFactory(defaultNotifFactory);
+        }
+    }
+
+    @ReactMethod
+    public void setAndroidLargeIcon(String iconName) {
+        int iconId = getImageResourceId(iconName);
+        if (iconId > 0) {
+            Preferences.getInstance().setAndroidLargeIconResourceId(iconId);
+            DefaultNotificationFactory defaultNotifFactory = getDefaultNotificationFactory();
+            defaultNotifFactory.setLargeIcon(iconId);
+            UAirship.shared().getPushManager().setNotificationFactory(defaultNotifFactory);
+        }
+    }
+
+    private DefaultNotificationFactory getDefaultNotificationFactory() {
+        final NotificationFactory notifFactory = UAirship.shared().getPushManager().getNotificationFactory();
```

Comment:
Sorry about this.... In our 8.0 release we moved all the setters to the base class, we can clean this up later.